### ISSUE TITLE
Allow throwing HttpResponseException from MVC (non WebApiCompatShim) controllers

### ISSUE THREAD

**Issue by pranavkm** [aspnet/Mvc#8704]
Based on discussion in https://github.com/aspnet/Mvc/issues/8570#issuecomment-431892499

**Comment by rynowak** [comment]
I don't think that there's a technical reason why we we couldn't do this, but redefining `HttpResponseException` inside the shared framework gives me pause. 

There's such a small amount of code here if we port the formatter.... Could this work as a sample?https://github.com/aspnet/Mvc/blob/master/src/Microsoft.AspNetCore.Mvc.WebApiCompatShim/HttpResponseExceptionActionFilter.cs

**Comment by pranavkm** [comment]
@glennc assigning this to you for specs. Let's try and figure this out in in a Design Review.
